src/agent_IR_RaymondWang_LinYang.py

```diff
@@ -0,0 +1,98 @@
+#trying
+""" Information Ratio """
+import argparse
+import numpy as np
+from collections import deque
+from pedlar.agent import Agent
+import time as tm
+
+
+
+class IRAgent(Agent):
+    name = "IR"
+    def __init__(self, verbose=False, **kwargs):
+        self.verbose=verbose
+        self.slow = deque(maxlen=128)
+        self.fast = deque(maxlen=32)
+        self.last_order = None
+        self.price_list=[]
+        self.std=0
+        self.total_profit_list=[]
+        self.total_profit=0
+        self.order_in_progress=False
+        super().__init__(**kwargs)
+
+
+    # print order when placing an order
+    def on_order(self, order):
+        """On order handler."""
+        print('----------------------------------------------')
+        print("ORDER: ", order)
+        self.order_in_progress=True
+        self.last_order=order.id
+        self.price_list=[]
+
+    # print the profit after the order is closed
+    def on_order_close(self, order, profit):
+        """On order close handler."""
+        print("PROFIT:", profit)
+        print('----------------------------------------------')
+        self.order_in_progress=False
+        self.total_profit+=profit
+        self.total_profit_list.append(self.total_profit)
+
+
+
+    def on_tick(self, bid, ask, time=None):
+        """On tick handler."""
+        print("Tick:", bid, ask)
+        if not (self.orders):
+            #print('no order')
+            self.slow.append(bid)
+            self.fast.append(bid)
+            if np.std(self.fast) == 0 or np.std(self.slow) == 0:
```

Comment:
store np.std(self.fast) in variable fast_std.
Do the same for np.std(self.slow)

Use fast_std and slow_std in the if statement

Reply:
Avoids it being calculated twice